### ISSUE TITLE
oauthInterceptor does not work for 401 (in my case)

### ISSUE THREAD

**Issue by ThaDafinser** [oauthjs/angular-oauth2#68]
When i log the `rejection` i get this result
![image](https://cloud.githubusercontent.com/assets/533017/10487911/88a723d0-7296-11e5-803b-859e247044d1.png)

The header from the API service is like this:
![image](https://cloud.githubusercontent.com/assets/533017/10487934/9cf7fddc-7296-11e5-91ef-4b0fe05017cc.png)

When i try the log the headers, they are `null` `console.log(rejection.headers("www-authenticate"));`

Am i doing something wrong or is the server response bad?

**Comment by anderson-rocha** [comment]
I had the same problem. But I solved it.

In the documentation, the oauthinterceptor works waiting these results:

{ status: 401, data: { error: 'invalid_token' }
{ status: 401, headers: { 'www-authenticate': 'Bearer realm="example"' } }

In his case, the data doesn't have the object with: error:'invalid_token'. 
In my case, the API that I'm using returns this object response: {error: 'access_denied'} . Then I changed the condition in angular-oauth2.js line 257:

```javascript
if (401 === rejection.status && (rejection.data && "access_denied" === rejection.data.error) || rejection.headers("www-authenticate") && 0 === rejection.headers("www-authenticate").indexOf("Bearer")) {
    $rootScope.$emit("oauth:error", rejection);
}
